Documentation/ContentBlocks/Carousel.rst

```diff
@@ -0,0 +1,182 @@
+:navigation-title: Carousel
+..  include:: /Includes.rst.txt
+
+..  _carousel:
+
+==============================
+Carousel Content Block example
+==============================
+
+This content block demonstrates some additional features of
+:composer:`friendsoftypo3/content-blocks`. We assume that you are already
+familiar with the concepts described in chapter
+`Custom Content Blocks <https://docs.typo3.org/permalink/t3sitepackage:content-blocks>`_.
+
+..  contents:: Table of contents
+
+..  _carousel-directory:
+
+Directory structure of the carousel content element
+===================================================
+
+Additionally to the files that the jumbotron (compare
+:ref:`content-blocks-jumbotron-directory`) the carousel comes with special CSS
+and JavaScript needed for this element only.
+
+Additionally it supplies a template for its display in the backend.
+
+..  directory-tree::
+    :level: 2
+    :show-file-icons: true
+
+    *   packages/my_site_package/ContentBlocks/ContentElements/jumbotron
+
+        *   assets
+
+            *   frontend.css
+            *   frontend.js
+            *   icon.svg
+
+        *   language
+
+            *   labels.xlf
+
+        *   templates
+
+            *   backend-preview.html
+            *   frontend.html
+
+        *   config.yaml
+
+..  _carousel-configuration:
+
+The carousel: A Content Block containing a collection of elements
+=================================================================
+
+The carousel can contain one or several items, each of which has an image,
+header, and description:
+
+..  literalinclude:: /CodeSnippets/my_site_package/ContentBlocks/ContentElements/carousel/config.yaml
+    :caption: packages/my_site_package/ContentBlocks/ContentElements/carousel/config.yaml
+    :linenos:
+
+Line 8: We use a field of type
+`Collection <https://docs.typo3.org/permalink/friendsoftypo3/content-blocks:field-type-collection>`_
+to contain the items to be displayed in the carousel. This field type expects
+an array of fields (line 10ff).
+
+Line 15: We use the type
```

Comment:
relationship manyToOne? Sure, we have a manyToOne relationship from sys_file to sys_file_reference, but we have a oneToOne relationship from sys_file_reference to tt_content. Damn, seems I really need to have a look into content blocks. Don't understand that part.

Reply:
manyToOne is correct.

A Content Element can have 0..n File References.
A File Reference belongs to exactly one Content Element.

tt_content 1 --- 0..n sys_file_reference

Reply:
Yes, it should be oneToOne. But as of now this makes technically no difference (but should be changed nevertheless)

Reply:
There are also some mistakes in TYPO3 Core then. They can also be fixed. Basically this is a new feature introduced with https://review.typo3.org/c/Packages/TYPO3.CMS/+/85046
When relationship is provided, your relations are resolved appropriately as a Collection (for oneToMany), which is the default or directly for anything else manyToOne, oneToOne.

Reply:
After this gets change in the site package builder we can change it here.